### PR TITLE
adding the possibility to handle multiples start/end/error loading handlers

### PR DESCRIPTION
**Pull request by biharck**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
It adds the possibility to handle multiple Start/Stop/Error Load Handlers.

* **What is the current behavior?** (You can also link to an open issue here)
There is no way to set more one handler for each viewport.

* **What is the new behavior (if this is a feature change)?**
The possibility to add more than one handler function

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: src/util/scrollToIndex.js [hunk 1]
```diff
@@ -44,9 +44,19 @@ export default function(element, newImageIdIndex) {
     newImageIdIndex += stackData.imageIds.length;
   }
 
-  const startLoadingHandler = loadHandlerManager.getStartLoadHandler();
-  const endLoadingHandler = loadHandlerManager.getEndLoadHandler();
-  const errorLoadingHandler = loadHandlerManager.getErrorLoadingHandler();
+  let viewportIndex = {};
```
Comment (swederik) [inline src/util/scrollToIndex.js]: Viewport index is only in OHIF. Can you use setStartLoadHandler(handler, element) where element is the actual DOM element? Not sure that works as a key, but I think it does.

If not we can use the enabled element's UUID: https://github.com/cornerstonejs/cornerstone/blob/master/src/enable.js#L69

**Comment by swederik** [comment]
You also need to add the removal of the handlers when the element is disabled.

File: src/util/scrollToIndex.js (hunk 1 above)
Comment (biharck) [reply src/util/scrollToIndex.js]: I tried with the DOM element but it didn't work. I'll try with UUID.

File: src/stateManagement/loadHandlerManager.js [hunk 2]
```diff
@@ -33,6 +78,7 @@ const loadHandlerManager = {
   getEndLoadHandler,
   setErrorLoadingHandler,
   getErrorLoadingHandler,
+  removeHandler,
```
Comment (swederik) [inline src/stateManagement/loadHandlerManager.js]: Could you make this `removeHandlers` (plural)?

File: src/store/internals/removeEnabledElement.js [hunk 3]
```diff
@@ -98,6 +100,17 @@ const _removeEnabledElement = function(enabledElement) {
   }
 };
 
+/**
+ * Remove load handler for the enabled element
+ * @private
+ * @method
+ * @param {HTMLElement} enabledElement
```
Comment (swederik) [inline src/store/internals/removeEnabledElement.js]: JSDoc is wrong, this should be `element`

File: src/stateManagement/loadHandlerManager.js (hunk 2 above)
Comment (biharck) [reply src/stateManagement/loadHandlerManager.js]: done

Comment (swederik) [reply src/stateManagement/loadHandlerManager.js]: It's still exporting the name as `removeHandler`....

File: src/store/internals/removeEnabledElement.js [hunk 4]
```diff
@@ -98,6 +100,17 @@ const _removeEnabledElement = function(enabledElement) {
...
+ * @returns {void}
+ */
+const _removeLoadHandlers = function(element) {
+  loadHandlerManager.removeHandler(element);
```
Comment (swederik) [inline src/store/internals/removeEnabledElement.js]: This needs to be updated to `removeHandlers` as well.

File: src/store/internals/removeEnabledElement.js (hunk 3 above)
Comment (biharck) [reply src/store/internals/removeEnabledElement.js]: done

File: src/store/internals/removeEnabledElement.js (hunk 4 above)
Comment (biharck) [reply src/store/internals/removeEnabledElement.js]: done

**Pull request merged**